Onboarding docs (#2678)
replicatedhq/replicated-docs: docs/vendor/replicated-onboarding.mdx

Author: This is the new onboarding topic that provides the main tasks users need to complete to onboard to the platform. The goal is to list out everything and provide links for more detailed information, rather than provide a "copy/paste" experience which is possible in something like a quick start/tutorial

The content is totally different from what was here before, so probably best to ignore the track changes

  Embedded Cluster has the following requirements:

  <Requirements/>

* (Optional) Ensure that you have kubectl access to a Kubernetes cluster. You will use this cluster to test installation with KOTS. If you do not intend to offer existing cluster installations with KOTS (for example, if you intend to support only Embedded Cluster and Helm installations for your users), then you do not need access to a cluster for the main onboarding tasks.

  You can use any cloud provider or tool that you prefer to create a cluster, such as [Replicated Compatibility Matrix](/vendor/testing-how-to), Google Kubernetes Engine (GKE), or minikube. 

## Onboard

Complete the tasks in this section to onboard your application. When you are done, you can continue to [Next Steps](#next-steps) to integrate other Replicated features with your application.

### Task 1: Create An Application

To get started with onboarding, first create a new application. This will be the official Vendor Portal application used by your team to create and promote both internal and customer-facing releases.

To create an application:

> 1. Create a new application using the Replicated CLI or the Vendor Portal. For the application name, use an official name for your application. You will not be able to modify the name. See [Create an Application](/vendor/vendor-portal-manage-app#create-an-application).

Author: Nice yeah I think it's worth digging in to the details here

To package your Helm chart with the Replicated SDK:

1. Go to the local directory where your Helm chart is.

1. In your application Helm chart `Chart.yaml` file, add the YAML below to declare the SDK as a dependency.
   
   If your application is installed as multiple charts, declare the SDK as a dependency of the chart that customers install first. Do not declare the SDK in more than one chart. For more information, see [Packaging a Helm Chart for a Release](helm-install-release).

   <DependencyYaml/>   

1. Update dependencies and package the chart as a `.tgz` file:

    <HelmPackage/>

    <UnauthorizedError/>

1. If your application is deployed as multiple Helm charts, package each chart as a separate `.tgz` archive using the `helm package -u PATH_TO_CHART` command. Do not declare the SDK in more than one chart.

### Task 4: Create the Initial Release with KOTS HelmChart and Embedded Cluster Config {#first-release}

> After packaging your Helm chart, you can create a release. The initial release for your application will include the minimum files required to install a Helm chart with the Embedded Cluster installer:

Author: It would be a larger change I think to move away from it. Which I am open to help kick off, but yeah it's a little pervasive

* The Helm chart `.tgz` archive
* [KOTS HelmChart custom resource](/reference/custom-resource-helmchart-v2)
* [Embedded Cluster Config](/reference/embedded-config)

If you have multiple charts, you will add each chart archive to the release, plus a corresponding KOTS HelmChart custom resource for each archive.

:::note
Configuring the KOTS HelmChart custom resource includes several tasks, and includes the use of KOTS template functions. Depending on how many Helm charts your application uses, Replicated recommends that you allow about two to three hours for configuring the HelmChart custom resource and creating and testing your initial release.
:::

To create the first release for your application:

1. In the local directory for your Helm chart, create a subdirectory named `manifests` where you will add the files for the release.

1. In the `manifests` directory:

   1. Move the `.tgz` chart archive that you packaged. If your application is deployed as multiple Helm charts, move each `.tgz` archive to `manifests`.

>    1. Create an `embedded-cluster.yaml` file with the following default Embedded Cluster Config:

Author: I think it's worth adding since it looks like I did add an explanation of the HelmChart custom resource in the next step below this one. 

Ended up putting some descriptions in those new <details> expand elements

      <EcCr/>
    
      For more information, see [Using Embedded Cluster](/vendor/embedded-overview).

>    1. Create a new YAML file. In this file, configure the KOTS HelmChart custom resource by completing the workflow in [Configuring the HelmChart Custom Resource](helm-native-v2-using).

Author: Made these updates. Also still need to update that topic to clarify the LocalRegistry template functions and how those work for EC installs

After successfully installing the initial release on a VM with Embedded Cluster (and optionally in an existing cluster with KOTS), go to the next task. You will continue to iterate throughout the rest of the onboarding process by creating and promoting new releases, then upgrading to the new version in your development environment.

### Task 5: Customize the KOTS Admin Console {#admin-console}

Configure the KOTS Application custom resource to add an application name, icon, and status informers. The name and icon will be displayed in the Admin Console and the Replicated Download Portal. The status informers will be used to display the application status on the Admin Console dashboard.

To configure the KOTS Application custom resource:

1. In your `manifests` directory, create a new `kots-app.yaml` file.

1. In the `kots-app.yaml` file, add the [KOTS Application](/reference/custom-resource-application) custom resource YAML and set the `title`, `icon`, and `statusInformers` fields.

   **Example:**

    ```yaml
    apiVersion: kots.io/v1beta1
    kind: Application
    metadata:
      name: gitea
    spec:
      title: Gitea

>       icon: https://raw.githubusercontent.com/cncf/artwork/master/projects/kubernetes/icon/color/kubernetes-icon-color.png

Author: Huh you know what I think that's a gap in the docs (I can't find anywhere we explain using a link versus base64 encoded image)

    apiVersion: kots.io/v1beta1
    kind: Config
    metadata:
      name: my-application
    spec:
      groups:
        - name: example_group
          title: Example Group
          items:
            - name: example_item
              title: Example Item
              type: text
              default: "Hello World"
    ```

    For more information, see:
    * [Creating and Editing Configuration Fields](/vendor/admin-console-customize-config-screen) 
    * [Using Conditional Statements in Configuration Fields](/vendor/config-screen-conditional)  
    * [Config](/reference/custom-resource-config)  

1. <CreateRelease/>

1. <TestYourChanges/>

1. In `manifests`, open the KOTS HelmChart custom resource that you configured in a previous step. Configure the `values` key of the HelmChart custom resource to map the fields in the KOTS Config custom resource to your Helm values.

   For more information, see:
   * [Mapping User-Supplied Values](/vendor/config-screen-map-inputs)
   * [Tutorial: Set Helm Chart Values with KOTS](/vendor/tutorial-config-setup)
   * [`values`](/reference/custom-resource-helmchart-v2#values) in _HelmChart v2_

1. <CreateRelease/>

1. <TestYourChanges/>

> 1. Continue to create and test new releases with new config fields until you are ready to move on to the next task.

Author: I updated the custom entitlements section in Next Steps to better explain this:
* I added subsections for "Map Values" and "Query Entitlements", which gives a little more space to explain both
* I also added a link to the "Mapping Helm Values with KOTS" topic in the new Map Values subsection, which does include an example of using KOTS template functions in the License context to map entitlements. (The example provided on that topic shows the use case of "this is a premium customer who gets access to a special feature")

        spec:
          collectors: []
          analyzers: []
    ```
    :::note
    If your application is installed as multiple Helm charts, you can optionally create separate support bundle specs in each chart. The specs are automatically merged a support bundle is generated. Alternatively, continue with a single support bundle spec and then optionally revisit how you organize your support bundle specs after you finish onboarding.
    :::
   
1. (Recommended) At a minimum, Replicated recommends that all support bundle specs include the `logs` collector. This collects logs from running Pods in the cluster.

   ```yaml
   apiVersion: v1
   kind: Secret
   metadata:
     name: example
     labels:
       troubleshoot.sh/kind: support-bundle
   stringData: 
     support-bundle-spec: |-
       apiVersion: troubleshoot.sh/v1beta2
       kind: SupportBundle
       metadata:
         name: example
       spec:
         collectors:
           - logs:
               selector:
                 - app.kubernetes.io/name=myapp
               namespace: {{ .Release.Namespace }}
               limits:
                 maxAge: 720h
                 maxLines: 10000

>    ```

Author: Oh yeah should be an example

   For more information, see:
   * <a href="/vendor/support-bundle-customizing">Adding and Customizing Support Bundles</a>
   * [Pod Logs](https://troubleshoot.sh/docs/collect/logs/) in the Troubleshoot documentation.

1. (Recommended) Ensure that any preflight checks that you added are also include in your support bundle spec. This ensures that support bundles collect at least the same information collected when running preflight checks. 

1. Update dependencies and package the chart as a `.tgz` file:

   <HelmPackage/>

1. Move the `.tgz` file to the `manifests` directory.

1. <CreateRelease/>

1. <TestYourChanges/>

   For information about how to generate support bundles, see [Generating Support Bundles](/vendor/support-bundle-generating).

1. (Optional) Customize the support bundle spec by adding additional collectors and analyzers. You can also choose to do this later. See [Step 2: Customize the Spec](/vendor/support-bundle-customizing#customize-the-spec) in _Adding and Customizing Support Bundles_.

### Task 9: Alias Replicated Endpoints with Your Own Domains

> Replicated recommends that you use custom domains to alias Replicated endpoints. Replicated domains are external to your domain and can require additional security reviews by your customer. Using custom domains as aliases can bring the domains inside an existing security review and reduce your exposure.

Reviewer: You can take it or leave it, and I'm not particular about the wording I wrote here. Just thinking about this explanation in a different way. I actually thought this would be a partial because I've seen this wording before, and I always felt like it could be worded more simply.
```suggestion
Your customers are exposed to several Replicated domains by default. Replicated recommends you use custom domains to unify the customer's experience with your brand and simplify security reviews.
```

Author: Nice, sounds good to me. And yeah this definitely appears in more than one place so I can see about adding it to a partial

For more information, see [Using Custom Domains](/vendor/custom-domains-using).

> ## Next Steps

Reviewer: EC Config supports roles for multi-node clusters. That could be a next step.

It also supports extensions, which might be necessary to deploy the application successfully if the vendor needs certain things deployed before their chart(s). So maybe that's not a next step and belongs higher up.

Author: Added both of these in (extensions as task 10)

Author: In the future we could also consider a more general "do more customizing to the EC Config" task as a Next Step

:::note
Before you test Helm installations for your application, you can complete the [Deploy a Helm Chart with KOTS and the Helm CLI](tutorial-kots-helm-setup) tutorial to learn how to install a single release with both KOTS and Helm.
:::

To support and test Helm installations:

1. Follow the steps in [Using the Proxy Registry with Helm Installations](/vendor/helm-image-registry) to authenticate with the Replicated proxy registry by creating a Secret with `type: kubernetes.io/dockerconfigjson` in your Helm chart.

1. Update dependencies and package the chart as a `.tgz` file:

    <HelmPackage/>

1. Add the `.tgz` file to a release. For more information, see [Managing Releases with the Vendor Portal](releases-creating-releases) or [Managing Releases with the CLI](releases-creating-cli).

1. Install the release in a cluster with the Helm CLI to test your changes. For more information, see [Installing with Helm](/vendor/install-with-helm).

> ### Add Support for Air Gap Installations

Author: Yeah I see what you're saying...I feel like I lean towards introducing as few caveats as possible to that Configuring HelmChart step since that topic is already a lot to follow...so that would mean if they already did the builder key above, then they can be pleasantly surprised down here that they are already part of the way done

KOTS and Embedded Cluster support installations in _air gap_ environments with no outbound internet access. Users can install in air gap environments by providing air gap bundles that contain the required images for the installers and for your application.

You can use the Vendor Portal to build and download the air gap bundles needed to install in an air gap existing cluster with KOTS or in an air gap VM or bare metal server with Embedded Cluster.

To add support for air gap installations:

> 1. If there are any images for your application that are not listed in your Helm chart, list these images in the `additionalImages` attribute of the KOTS Application custom resource. This ensures that the images are included in the `.airgap` bundle for the release. See [Define Additional Images](/vendor/operator-defining-additional-images).

Author: Yeah that would be helpful--I can include the Operator example and if you think of any others lmk

1. If you have any Helm values that determine whether or not an application image is included in an installation, then configure the `builder` key in the KOTS HelmChart custom resource to enable those values. The Vendor Portal uses the values provided in the `builder` key to build the `.airgap` bundle for the release. See [Packaging Air Gap Bundles for Helm Charts](/vendor/helm-packaging-airgap-bundles).

    :::note
    If the default values in your Helm chart already enable all the images needed to successfully deploy the chart, then you do not need to configure the `builder` key.
    :::

> 1. If you have not done so already as part of [Task 4: Create and Install the Initial Release](#first-release), ensure that the `values` key in the KOTS HelmChart custom resource correctly rewrites image names for air gap installations. This is done using the KOTS HasLocalRegistry, LocalRegistryHost, and LocalRegistryNamespace template functions to render the location of the given image on the user's own local registry.

Reviewer: ```suggestion
1. If you have not done so already as part of [Task 4: Create and Install the Initial Release](#first-release), ensure that the `values` key in the KOTS HelmChart custom resource correctly rewrites image names for air gap installations. This is done using the KOTS HasLocalRegistry, LocalRegistryHost, and LocalRegistryNamespace template functions to render the location of the given image in the user's own local registry.
```

> 1. If you have not done so already as part of [Task 4: Create and Install the Initial Release](#first-release), ensure that the `values` key in the KOTS HelmChart custom resource correctly rewrites image names for air gap installations. This is done using the KOTS HasLocalRegistry, LocalRegistryHost, and LocalRegistryNamespace template functions to render the location of the given image on the user's own local registry.

Reviewer: "user's own local registry" might be misleading since this does also apply to the registry included with EC

> 1. If you have not done so already as part of [Task 4: Create and Install the Initial Release](#first-release), ensure that the `values` key in the KOTS HelmChart custom resource correctly rewrites image names for air gap installations. This is done using the KOTS HasLocalRegistry, LocalRegistryHost, and LocalRegistryNamespace template functions to render the location of the given image on the user's own local registry.

Author: >"user's own local registry" might be misleading since this does also apply to the registry included with EC

True. Also still need to update that info in the HelmChart v2 topic to explain how the localregistry template functions will pick up on the registry used by EC

> 1. If you have not done so already as part of [Task 4: Create and Install the Initial Release](#first-release), ensure that the `values` key in the KOTS HelmChart custom resource correctly rewrites image names for air gap installations. This is done using the KOTS HasLocalRegistry, LocalRegistryHost, and LocalRegistryNamespace template functions to render the location of the given image on the user's own local registry.

Reviewer: It applies to the kURL registry too FYI

> 1. If you have not done so already as part of [Task 4: Create and Install the Initial Release](#first-release), ensure that the `values` key in the KOTS HelmChart custom resource correctly rewrites image names for air gap installations. This is done using the KOTS HasLocalRegistry, LocalRegistryHost, and LocalRegistryNamespace template functions to render the location of the given image on the user's own local registry.

Author: Ah yeah and kurl. I've been pretending like kurl doesn't exist

   For more information, see [Rewrite Image Names](/vendor/helm-native-v2-using#rewrite-image-names) in _Configuring the HelmChart Custom Resource v2_.

1. Create and promote a new release with your changes. For more information, see [Managing Releases with the Vendor Portal](releases-creating-releases) or [Managing Releases with the CLI](releases-creating-cli).

1. In the [Vendor Portal](https://vendor.replicated.com), go the channel where the release was promoted to build the air gap bundle. Do one of the following:
     * If the **Automatically create airgap builds for newly promoted releases in this channel** setting is enabled on the channel, watch for the build status to complete.
     * If automatic air gap builds are not enabled, go to the **Release history** page for the channel and build the air gap bundle manually.
     

>      For more information, see [Downloading Air Gap Bundles](/vendor/releases-download-airgap-bundles#air-gap-bundle).

Author: Ah yeah this is going to be out of date now. Just merged that PR to remove this topic so will have to go back through and update this

1. Create a customer with the **Airgap Download Enabled** entitlement enabled so that you can test air gap installations. See [Creating and Managing Customers](/vendor/releases-creating-customer).

> 1. Download the Embedded Cluster release artifacts for air gap installations, then install with Embedded Cluster on an air gap VM to test. See [Installing in Air Gap Environments with Embedded Cluster](/enterprise/installing-embedded-air-gap).

Author: Good call--adding a story in case I don't get to it as part of this pr https://app.shortcut.com/replicated/story/113898/clarify-that-air-gap-license-is-required-for-ec-install

You can also configure the Kubernetes SIG Application resource add links to other resources like documentation or dashboards.

For more information, see [Adding Application Links to the Dashboard](/vendor/admin-console-adding-buttons-links).

### Update the Preflight and Support Bundles Specs

After adding basic specs for preflights and support bundles, you can continue to add more collectors and analyzers as needed.

Consider the following recommendations and best practices:

* Revisit your preflight and support bundle specs when new support issues arise that are not covered by your existing specs.

* Your support bundles should include all of the same collectors and analyzers that are in your preflight checks. This ensures that support bundles include all the necessary troubleshooting information, including any failures in preflight checks.

* Your support bundles will most likely need to include other collectors and analyzers that are not in your preflight checks. This is because some of the information used for troubleshooting (such as logs) is not necessary when running preflight checks before installation.

> * If your application is installed as multiple Helm charts, you can optionally add separate support bundle specs in each chart. This can make it easier to keep the specs up-to-date and to avoid merge conflicts that can be caused when multiple team members contribute to a single, large support bundle spec. When an application has multiple support bundle specs, the specs are automatically merged when generating a support bundle so that only a single support bundle archive is provided to the user.

Author: Yeah you're right, "bundle" is already an "archive"

The documentation for the open-source Troubleshoot project includes the full list of available collectors and analyzers that you can use. See [All Collectors](https://troubleshoot.sh/docs/collect/all/) and the [Analyze](https://troubleshoot.sh/docs/analyze/) section in the Troubleshoot documentation.

You can also view common examples of collectors and analyzers used in preflight checks and support bundles in [Preflight Spec Examples](preflight-examples) and [Support Bundle Spec Examples](support-bundle-examples).

### Add and Query Custom License Entitlements

> You can add custom license entitlements for your application in the Vendor Portal. Custom license fields are useful when there is entitlement information that applies to a subset of customers. For example, you can create a custom license field to limit the number of active users permitted. Or, you can create a field that limits the number of nodes a customer is permitted on their cluster.

Reviewer: I mentioned above that license fields may be used in HelmChart values too (like if someone is on the premium plan, and some extra service gets deployed as part of that)

For more information, see:
* [Managing Custom License Fields](/vendor/licenses-adding-custom-fields)
* [Querying Entitlements with the Replicated SDK API](/vendor/licenses-reference-sdk)
* [Checking Entitlements in Preflights with KOTS Template Functions](/vendor/licenses-referencing-fields)
* [Verifying License Field Signatures with the Replicated SDK API](/vendor/licenses-verify-fields-sdk-api)
* [Creating and Managing Customers](/vendor/releases-creating-customer)
* [Developing Against the SDK API](/vendor/replicated-sdk-development)

### Configure Backup and Restore

Enable backup and restore with Velero for your application so that users can back up and restore their KOTS Admin Console and application data. 

There are different steps to configure backup and restore for Embedded Cluster and for existing cluster installations with KOTS:
* To configure the disaster recovery feature for Embedded Cluster, see [Disaster Recovery for Embedded Cluster](/vendor/embedded-disaster-recovery)
* To configure the snapshots feature for existing cluster KOTS installations, see [Configuring Backup and Restore](snapshots-configuring-backups).

### Add Custom Metrics

> In addition to the built-in insights displayed in the Vendor Portal by default (such as uptime and time to install), you can also configure custom metrics to measure instances of your application running customer environments. Custom metrics can be collected for application instances running in online or air gap environments using the Replicated SDK.

Reviewer: ```suggestion
In addition to the built-in insights displayed in the Vendor Portal by default (such as uptime and time to install), you can also configure custom metrics to measure instances of your application running in customer environments. Custom metrics can be collected for application instances running in online or air gap environments using the Replicated SDK.
```

For more information, see:
* [About Integrating with CI/CD](/vendor/ci-overview)
* [About Compatibility Matrix](/vendor/testing-about)
* [Recommended CI/CD Workflows](/vendor/ci-workflows)

### Customize Release Channels

By default, the Vendor Portal includes Unstable, Beta, and Stable channels. You can customize the channels in the Vendor Portal based on your application needs.

Consider the following recommendations:
* Use the Stable channel for your primary release cadence. Releases should be promoted to the Stable channel only as frequently as your average customer can consume new releases. Typically, this is no more than monthly. However, this cadence varies depending on the customer base.
* If you have a SaaS product, you might want to create an "Edge" channel where you promote the latest SaaS releases.
* You can consider a “Long Term Support” channel where you promote new releases less frequently and support those releases for longer.

> * It can be useful to create channels for each feature branch so that internal teams reviewing a PR can easily get the installation artifacts as well as review the code. You can automate channel creation as part of a pipeline or Makefile.

Reviewer: I wonder if we have a documented flow for this. I remember some automation around this with the `--auto` flag for `replicated release create`. Just wondering if some docs on how this can be accomplished would help.

Author: Agreed, story added for that: https://app.shortcut.com/replicated/story/113941/explain-how-to-create-channels-during-release-creation-with-auto-flag

1. Update dependencies and package the chart as a `.tgz` file:

    <HelmPackage/>

1. Add the `.tgz` file to a release. For more information, see [Managing Releases with the Vendor Portal](releases-creating-releases) or [Managing Releases with the CLI](releases-creating-cli).

1. Install the release in a cluster with the Helm CLI to test your changes. For more information, see [Installing with Helm](/vendor/install-with-helm).

### Add Support for Air Gap Installations

KOTS and Embedded Cluster support installations in _air gap_ environments with no outbound internet access. Users can install in air gap environments by providing air gap bundles that contain the required images for the installers and for your application.

You can use the Vendor Portal to build and download the air gap bundles needed to install in an air gap existing cluster with KOTS or in an air gap VM or bare metal server with Embedded Cluster.

To add support for air gap installations:

> 1. If there are any images for your application that are not listed in your Helm chart, list these images in the `additionalImages` attribute of the KOTS Application custom resource. This ensures that the images are included in the air gap bundle for the release. One common use case for this is applications that use Kubernetes Operators, which might have images listed in objects created by the Operator and not in the Helm chart. See [Define Additional Images](/vendor/operator-defining-additional-images).

Author: >which might have images listed in objects created by the Operator and not in the Helm chart.

not sure if this is the best phrasing

import UnauthorizedError from "../partials/replicated-sdk/_401-unauthorized.mdx"

# Replicated Onboarding 

This topic describes how to onboard applications to the Replicated Platform.

## Before You Begin

This section includes guidance and prerequisites to review before you begin onboarding your application.  

### Best Practices and Recommendations

The following are some best practices and recommendations for successfully onboarding with Replicated:

* When integrating new Replicated features with an application, make changes in small iterations and test frequently by installing or upgrading the application in a development environment. This will help you to more easily identify issues and troubleshoot. This onboarding workflow will guide you through the process of integrating features in small iterations.

* Use the Replicated CLI to create and manage your application and releases. Getting familiar with the Replicated CLI will also help later on when integrating Replicated workflows into your CI/CD pipelines. For more information, see [Installing the Replicated CLI](/reference/replicated-cli-installing).

> * These onboarding tasks assume that you will test the installation of each release on a VM with the Replicated Embedded Cluster installer _and_ in a cluster with the Replicated KOTS installer. If you do not intend to offer existing cluster installations with KOTS (for example, if you intend to support only Embedded Cluster and Helm installations for your users), then can choose to test with Embedded Cluster only.

Author: ^ I've framed the KOTS existing cluster install as optional. Not sure if that's the direction to go, but my thought was that it would at least make onboarding quicker if they just test EC installs, and the releases themselves will also work for KOTS existing cluster installs anyway, so it's not like they'll have to change anything if they change their minds

(Note that there's a separate task in the Next Steps section that explains how to add support for Helm installs and test, so that's why that's not called out here explicitly)

    <UnauthorizedError/>

1. If your application is deployed as multiple Helm charts, package each chart as a separate `.tgz` archive using the `helm package -u PATH_TO_CHART` command. Do not declare the SDK in more than one chart.

### Task 4: Create the Initial Release with KOTS HelmChart and Embedded Cluster Config {#first-release}

After packaging your Helm chart, you can create a release. The initial release for your application will include the minimum files required to install a Helm chart with the Embedded Cluster installer:
* The Helm chart `.tgz` archive
* [KOTS HelmChart custom resource](/reference/custom-resource-helmchart-v2)
* [Embedded Cluster Config](/reference/embedded-config)

If you have multiple charts, you will add each chart archive to the release, plus a corresponding KOTS HelmChart custom resource for each archive.

:::note
Configuring the KOTS HelmChart custom resource includes several tasks, and includes the use of KOTS template functions. Depending on how many Helm charts your application uses, Replicated recommends that you allow about two to three hours for configuring the HelmChart custom resource and creating and testing your initial release.

> :::

Author: ^ Added a heads up about the time it'll take to configure the HelmChart custom resource. Not sure how accurate this estimate is

1. In the local directory for your Helm chart, create a subdirectory named `manifests` where you will add the files for the release.

1. In the `manifests` directory:

   1. Move the `.tgz` chart archive that you packaged. If your application is deployed as multiple Helm charts, move each `.tgz` archive to `manifests`.

   1. Create an `embedded-cluster.yaml` file with the following default Embedded Cluster Config:

      <EcCr/>
    
      For more information, see [Using Embedded Cluster](/vendor/embedded-overview).

   1. Create a new YAML file. In this file, configure the KOTS HelmChart custom resource by completing the workflow in [Configuring the HelmChart Custom Resource](helm-native-v2-using).
   
      The KOTS HelmChart custom resource is required to install Helm charts with KOTS and Embedded Cluster. As part of configuring the KOTS HelmChart custom resource, you will rewrite image names and add image pull secrets to allow your application images to be accessed through the Replicated proxy registry.

   1. If your application is deployed as multiple Helm charts, repeat the step above to add a separate HelmChart custom resource for each Helm chart archive in the release.

>    1. If there any values in any of your Helm charts that need to be set for the installation to succeed, you can set those values using the `values` key in the corresponding HelmChart custom resource. See [Setting Helm Values with KOTS](/vendor/helm-optional-value-keys).

Author: ^ Added this step about making sure your Helm values will allow for a successful install. Can take it out if we think it's not necessary

   
      This is a temporary measure to ensure the values get passed to the Helm chart during installation until you configure the Admin Console Config screen in a later onboarding task. If your default Helm values are sufficient for installation, you can skip this step.  

> 

Author: ^ ^ Added this step about making sure your Helm values will allow for a successful install. Can take it out if we think it's not necessary

1. If you have any Helm values that determine whether or not an application image is included in an installation, then configure the `builder` key in the KOTS HelmChart custom resource to enable those values. The Vendor Portal uses the values provided in the `builder` key to build the `.airgap` bundle for the release. See [Packaging Air Gap Bundles for Helm Charts](/vendor/helm-packaging-airgap-bundles).

    :::note
    If the default values in your Helm chart already enable all the images needed to successfully deploy the chart, then you do not need to configure the `builder` key.
    :::

1. If you have not done so already as part of [Task 4: Create and Install the Initial Release](#first-release), ensure that the `values` key in the KOTS HelmChart custom resource correctly rewrites image names for air gap installations. This is done using the KOTS HasLocalRegistry, LocalRegistryHost, and LocalRegistryNamespace template functions to render the location of the given image on the user's own local registry.

>    For more information, see [Rewrite Image Names](/vendor/helm-native-v2-using#rewrite-image-names) in _Configuring the HelmChart Custom Resource v2_.

Author: ^ This is probable unrelated to this PR in particular, but the HasLocalRegistry stuff I think is just for existing cluster KOTS installs, and not necessary for Embedded Cluster installs, right? If so, we should probably update the HelmChart topic to explain this and explain how images get rewritten for EC installs

Reviewer: It's necessary for both. The registry deployed as part of EC is returned by the local registry functions, so that's how the images get pulled from the EC registry.
